core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -97,4 +108,35 @@ private ProbeResult fromSCMLink(ProbeContext context, String scm) {
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    public static String searchPomFiles(File directory, String pluginName) {
+        File[] files = directory.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    searchPomFiles(file, pluginName);
+                } else {
+                    return getSCMFolderPath(file.getAbsolutePath(), pluginName);
+                }
+            }
+        }
+        return null;
+    }
+
+    public static String getSCMFolderPath(String filePath, String pluginName) {
+        if (filePath.startsWith("pom.xml")) {
+            MavenXpp3Reader mavenReader = new MavenXpp3Reader();
+            try (Reader reader = new InputStreamReader(new FileInputStream(filePath), StandardCharsets.UTF_8)) {
+                Model model = mavenReader.read(reader);
+                if (model.getPackaging().equals("hpi") && model.getArtifactId().equals(pluginName)) {
+                    return filePath + "/pom.xml";
```

Comment:
This is not returning the folder path, but the `pom.xml` path, which is different from what you were retrieving with the previous version of the code.

Reply:
I'm not sure we want this code to run on Windows based systems, but I'd recommend not hardcoding path separators and not using String for paths. Rather use the NIO all around and pass paths as `Path` objects, which have methods to resolve path in an OS independent way.

Reply:
> This is not returning the folder path, but the `pom.xml` path,

So I should return the value of the `artifact id`? Because that is the folder path, right? The name of the repo and the folder?

Reply:
The folder path is everything after the `repo` that leads to? Pom file?